source/ShipEffectsShader.cpp

```diff
@@ -0,0 +1,370 @@
+	GLint transformI;
+	GLint blurI;
+	GLint clipI;
+
+	GLint recentHitsCountI;
+	GLint recentDamageI;
+	GLint recentHitsI;
+	GLint shieldColorI;
+	GLint ratioI;
+	GLint sizeI;
+
+	GLint fastI;
+
+	GLuint vao;
+	GLuint vbo;
+}
+
+Point ShipEffectsShader::center = Point();
+
+// Initialize the shaders.
+void ShipEffectsShader::Init()
+{
+
+	static const char *vertexCode =
+		"// vertex sprite shader\n"
+		"precision mediump float;\n"
+		"uniform vec2 scale;\n"
+		"uniform vec2 position;\n"
+		"uniform mat2 transform;\n"
+		"uniform vec2 blur;\n"
+		"uniform float clip;\n"
+
+		"in vec2 vert;\n"
+		"out vec2 fragTexCoord;\n"
+		"out vec2 shrinkby;\n"
+
+		"void main() {\n"
+		"  vec2 blurOff = 2.f * vec2(vert.x * abs(blur.x), vert.y * abs(blur.y));\n"
+		"  gl_Position = vec4((transform * (vert + blurOff) + position) * scale, 0, 1);\n"
+		"  vec2 texCoord = vert + vec2(.5, .5);\n"
+		"  shrinkby = scale;\n"
+		"  fragTexCoord = vec2(texCoord.x, min(clip, texCoord.y)) + blurOff;\n"
+		"}\n";
+
+	static const char *fragmentCode =
+		"// fragment sprite shader\n"
+		"precision mediump float;\n"
+#ifdef ES_GLES
+		"precision mediump sampler2DArray;\n"
+#endif
+		"uniform sampler2DArray tex;\n"
+		"uniform float frame;\n"
+		"uniform float frameCount;\n"
+		"uniform vec2 blur;\n"
+		"const int range = 5;\n"
+
+		"uniform vec2 recentHits[64];\n"
+		"uniform float recentDamage[64];\n"
+		"uniform int recentHitCount;\n"
+		"uniform vec4 shieldColor;\n"
+		"uniform float ratio;\n"
+		"uniform float size;\n"
+
+		"uniform int isFast;\n"
+
+		"in vec2 fragTexCoord;\n"
+		"in vec2 shrinkby;\n"
+
+		"out vec4 finalColor;\n"
+
+		"float first = floor(frame);\n"
+		"float second = mod(ceil(frame), frameCount);\n"
+		"float fade = frame - first + second;\n"
+
+		"vec4 sampleSmooth(sampler2DArray sampler, vec2 uv)\n"
+		"{\n"
+		"  return mix( texture(tex, vec3(uv, first)),\n"
+		"    texture(tex, vec3(uv, second)), fade);\n"
+		"}\n"
+
+		"float stripe(float a, float mod)\n"
+		"{\n"
+		"  return clamp(sin(a*size*4.) * 2. + mod, 0., 1.);"
+		"}\n"
+
+		"float sobellish(vec2 uv)\n"
+		"{\n"
+		"  float obel = 0.;\n"
+		"  for (int x = -3; x <= 3; x++)\n"
+		"  {\n"
+		"    for (int y = -3; y <= 3; y++)\n"
+		"	 {\n"
+		"      obel += sampleSmooth(tex, uv + vec2(x, y) / (300.)).a;\n"
+		"    }\n"
+		"  }\n"
+		"  obel /= 49.;\n"
+		"  return sqrt(2. * obel + 0.2 / (obel / 2. - .6) + 0.3);\n"
+		"}\n"
+
+		"float gridPattern(float f, vec2 uv)\n"
+		"{\n"
+		"  return f + (f - f * (stripe(uv.x * ratio, 1.5) * stripe(uv.y, 1.5)));\n"
+		"}\n"
+
+		"void main()\n"
+		"{\n"
+		"  vec2 uv = fragTexCoord;\n"
+		"  vec4 color;"
+		"  if(isFast == 0)\n"
+		"  {\n"
+		"    float totalimpact = 1.;"
+		"    for(int i = 0; i < recentHitCount; i++)\n"
+		"    {\n"
+		"      vec2 hitPoint = recentHits[i] + vec2(0.5, 0.5);\n"
+		"      totalimpact += recentDamage[i];"
+		"      color += shieldColor * recentDamage[i] * clamp(2. - distance(hitPoint, uv)*.04*size, 0., 1.5);\n"
```

Comment:
```suggestion
		"      color += shieldColor * recentDamage[i] * clamp(2. - distance(hitPoint, uv)*.04*80, 0., 1.5);\n"
```

Instead of `item.size = 80` (from my earlier feedback) you could update the shader to hard-code it

Reply:
not here